pkg/restore/restore.go

```diff
@@ -510,6 +540,52 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 	return warnings, errs
 }
 
+func (ctx *context) createNamespaceAfterTerminate(ns *v1.Namespace, logger *logrus.Entry) error {
+	timeout := time.NewTimer(ctx.resourceTerminatingTimeout)
+	defer timeout.Stop()
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-timeout.C:
+			return errors.Errorf("Timed out waiting on namespace %s to be ready", ns.Name)
+		case <-ticker.C:
+			c, err := ctx.namespaceClient.Create(ns)
+			if apierrors.IsAlreadyExists(err) && c.Status.Phase == v1.NamespaceTerminating {
+				logger.Debugf("Namespace %s still terminating, waiting", ns.Name)
+			} else if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			} else {
+				return nil
+			}
+
+		}
+	}
+}
+
+func (ctx *context) waitForPVToTerminate(pv *v1.PersistentVolume) error {
+	timeout := time.NewTimer(ctx.resourceTerminatingTimeout)
+	defer timeout.Stop()
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-timeout.C:
+			return errors.Errorf("Timed out waiting on PV %s to be deleted", pv.Name)
+		case <-ticker.C:
+			pv, err := ctx.pvClient.Get(pv.Name, metav1.GetOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+			if isPVReleasedOrBound(pv) {
```

Comment:
I think this should be `if err == nil`, right? we should continue waiting in all cases where there's no error getting the PV, i.e. it still exists?